docker-storage-setup.sh

```diff
@@ -707,32 +707,56 @@ scan_disks() {
 }
 
 create_partition_sfdisk(){
-  local dev="$1" size
+  local dev="$1" size label="$2" type
   # Use a single partition of a whole device
   # TODO:
-  #   * Consider gpt, or unpartitioned volumes
+  #   * Consider unpartitioned volumes
   #   * Error handling when partition(s) already exist
   #   * Deal with loop/nbd device names. See growpart code
+
   size=$(( $( awk "\$4 ~ /"$( basename $dev )"/ { print \$3 }" /proc/partitions ) * 2 - 2048 ))
+  # if parted is not present, and this version of sfdisk
+  # does not support GPT then we are limited to 2TB
+  if [ $label = "gpt" ]; then
+    if $(sfdisk --help|grep gpt >/dev/null) ; then
+      label='label: gpt'
+      type=''
+      size=$(( $size - 34 ))
+    else
+      Info "Your version of sfdisk does not support GPT labels. Will use MBR instead but partition size will be limited to 2TB"
+      label=''
+      type=', Id=8e'
+      # 2TB in sectors - 2048
+      if [ $size -gt 4294965248 ]; then size=4294965248; fi
+    fi
+  elif [ $label = "msdos" ]; then
+    label=''
+  fi
+  
     cat <<EOF | sfdisk $dev
 unit: sectors
+$label
 
-${dev}1 : start=     2048, size=  ${size}, Id=8e
+${dev}1 : start=     2048, size=  ${size}${type}
 EOF
 }
 
 create_partition_parted(){
-  local dev="$1"
-  parted $dev --script mklabel msdos mkpart primary 0% 100% set 1 lvm on
+  local dev="$1" label="$2"
+  parted $dev --script mklabel $label mkpart primary 0% 100% set 1 lvm on
```

Comment:
On this note, when using `sfdisk` no name seems to be set:

```
(parted) print                                                            
Model: ATA VBOX HARDDISK (scsi)
Disk /dev/sdb: 8590MB
Sector size (logical/physical): 512B/512B
Partition Table: gpt
Disk Flags: 

Number  Start   End     Size    File system  Name  Flags
 1      1049kB  8590MB  8589MB
```

When trying to recreate the partition with parted 3.1 (included in CentOS 7) or parted 3.2 (Fedora 25/cloud-base), `mkpart` seems to expect a non-empty string for name but only when running in non-interactive mode:

```
(parted) rm 1
(parted) q
[root@localhost ~]# parted /dev/sdb --script mkpart "" 0% 100% 
[root@localhost ~]# parted /dev/sdb
(parted) print                                                            
Model: ATA VBOX HARDDISK (scsi)
Disk /dev/sdb: 8590MB
Sector size (logical/physical): 512B/512B
Partition Table: gpt
Disk Flags: 

Number  Start  End  Size  File system  Name  Flags

(parted) mkpart                                                           
Partition name?  []?                                                      
File system type?  [ext2]?                                                
Start? 2048                                                               
End? -1                                                                   
(parted) print                                                            
Model: ATA VBOX HARDDISK (scsi)
Disk /dev/sdb: 8590MB
Sector size (logical/physical): 512B/512B
Partition Table: gpt
Disk Flags: 

Number  Start   End     Size    File system  Name  Flags
 1      2048MB  8589MB  6541MB

(parted)
```

I find this inconsistency quite annoying, but it does mean that we will need a partition name when using parted.